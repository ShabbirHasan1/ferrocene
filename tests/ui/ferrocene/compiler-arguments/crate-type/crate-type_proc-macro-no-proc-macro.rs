// check-fail
// compile-flags: --crate-type=proc-macro
// needs-dynamic-linking - proc macros rely on dynamic linking

extern crate proc_macro;

use proc_macro::TokenStream;

pub fn foo(input: TokenStream) -> TokenStream { //~ ERROR `proc-macro` crate
    input
}

// ferrocene-annotations: um_rustc_crate_type
