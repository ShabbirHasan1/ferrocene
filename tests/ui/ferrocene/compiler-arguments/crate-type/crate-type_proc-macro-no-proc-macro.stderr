error: `proc-macro` crate types currently cannot export any items other than functions tagged with `#[proc_macro]`, `#[proc_macro_derive]`, or `#[proc_macro_attribute]`
  --> $DIR/crate-type_proc-macro-no-proc-macro.rs:9:1
   |
LL | pub fn foo(input: TokenStream) -> TokenStream {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

